packages/containers/src/List/List.sagas.js

```diff
@@ -0,0 +1,76 @@
+import { put, takeEvery, take, call } from 'redux-saga/effects';
+import Connected from './List.connect';
+import Constants from './List.constant';
+import { getCollectionItems, sortList, filterList } from './selector';
+
+export function* updateList(data) {
+	const collectionId = data.props.collectionId || 'default';
+	const state = data.context.store.getState();
+	const componentState = state.cmf.components.getIn(['Container(List)', collectionId]);
+	const items = getCollectionItems(state, collectionId);
+
+	let results = items || data.props.config.items;
+	results = sortList(componentState, results, data.props.config);
+	results = filterList(componentState, results, data.props.config);
+	yield put(
+		Connected.setStateAction(
+			{
+				items: results,
+			},
+			collectionId,
+		),
+	);
+}
+
+export function* onFilterChange(data) {
+	yield put(
+		Connected.setStateAction(
+			{
+				searchQuery: data.payload.query,
+			},
+			data.props.collectionId || 'default',
+		),
+	);
+	yield call(updateList, data);
+}
+
+export function* onToggleFilter(data) {
+	const filterData = data;
```

Comment:
you can change the arg name or remove this variable

Reply:
the data needs to be updated to invoke filter change when the filterBar is closed and reassignment is not allowed here so using the variable

Reply:
All objects are given by reference so when you're assigning `data` to `filterData`, you just getting the reference, not a copy.
So below, writing `filterData.payload.searchQuery = '';` or `data.payload.searchQuery = '';` is equivalent.